functests/1_performance/performance.go

```diff
@@ -340,8 +342,134 @@ var _ = Describe("[rfe_id:27368][performance]", func() {
+			err = testclient.GetWithRetry(context.TODO(), key, v1Profile)
+			ExpectWithOffset(1, err).ToNot(HaveOccurred(), "Failed getting v1profile")
+			Expect(verifyV1alpha1Conversion(v1alpha1Profile, v1Profile)).ToNot(HaveOccurred())
+
+			Expect(testclient.Client.Delete(context.TODO(), v1Profile)).ToNot(HaveOccurred())
+		})
+	})
 })
 
+func verifyV1alpha1Conversion(v1alpha1Profile *performancev1alpha1.PerformanceProfile, v1Profile *performancev1.PerformanceProfile) error {
+	specCPU := v1alpha1Profile.Spec.CPU
+	if (specCPU == nil) != (v1Profile.Spec.CPU == nil) {
+		return fmt.Errorf("spec CPUs field is different")
+	}
+
+	if specCPU != nil {
+		if string(*specCPU.Reserved) != string(*v1Profile.Spec.CPU.Reserved) {
+			return fmt.Errorf("reserved CPUs are different [v1alpha1: %s, v1: %s]",
+				*specCPU.Reserved, *v1Profile.Spec.CPU.Reserved)
+		}
+
+		if string(*specCPU.Isolated) != string(*v1Profile.Spec.CPU.Isolated) {
+			return fmt.Errorf("isolated CPUs are different [v1alpha1: %s, v1: %s]",
+				*specCPU.Isolated, *v1Profile.Spec.CPU.Isolated)
+		}
+
+		if specCPU.BalanceIsolated != v1Profile.Spec.CPU.BalanceIsolated {
+			return fmt.Errorf("balanceIsolated field is different [v1alpha1: %t, v1: %t]",
+				*specCPU.BalanceIsolated, *v1Profile.Spec.CPU.BalanceIsolated)
+		}
+	}
+
+	specHugePages := v1alpha1Profile.Spec.HugePages
+	if (specHugePages == nil) != (v1Profile.Spec.HugePages == nil) {
+		return fmt.Errorf("spec HugePages feild is different")
+	}
+
+	if specHugePages != nil {
+		if string(*specHugePages.DefaultHugePagesSize) != string(*v1Profile.Spec.HugePages.DefaultHugePagesSize) {
+			return fmt.Errorf("defaultHugePagesSize field is different [v1alpha1: %s, v1: %s]",
+				*specHugePages.DefaultHugePagesSize, *v1Profile.Spec.HugePages.DefaultHugePagesSize)
+		}
+
+		if (specHugePages.Pages == nil) != (v1Profile.Spec.HugePages.Pages == nil) ||
+			len(specHugePages.Pages) != len(v1Profile.Spec.HugePages.Pages) {
+			return fmt.Errorf("pages field is different [v1alpha1: %v, v1: %v]",
+				specHugePages.Pages, v1Profile.Spec.HugePages.Pages)
+		}
+
+		for i, v1alpha1Page := range specHugePages.Pages {
+			v1page := v1Profile.Spec.HugePages.Pages[i]
+			if string(v1alpha1Page.Size) != string(v1page.Size) ||
+				(v1alpha1Page.Node == nil) != (v1page.Node == nil) ||
+				(v1alpha1Page.Node != nil && *v1alpha1Page.Node != *v1page.Node) ||
+				v1alpha1Page.Count != v1page.Count {
+				return fmt.Errorf("pages field is different [v1alpha1: %v, v1: %v]",
+					specHugePages.Pages, v1Profile.Spec.HugePages.Pages)
+			}
+		}
+	}
+
+	if !reflect.DeepEqual(v1alpha1Profile.Spec.MachineConfigLabel, v1Profile.Spec.MachineConfigLabel) {
+		return fmt.Errorf("machineConfigLabel field is different [v1alpha1: %v, v1: %v]",
+			v1alpha1Profile.Spec.MachineConfigLabel, v1Profile.Spec.MachineConfigLabel)
+	}
+
+	if !reflect.DeepEqual(v1alpha1Profile.Spec.MachineConfigPoolSelector, v1Profile.Spec.MachineConfigPoolSelector) {
```

Comment:
can we use DeepEqual (or go-cmp?) for the previous fields as well?

Reply:
As far as I understood DeepEqual checks types an values. In our case the sub-structures have different types so it can't be used( v1alpha1Profile.Spec.CPU type vs v1Profile.Spec.CPU). go-cmp it seems it asks for a "comparer" function, so nothing is really saved.

Point is even if the structures and substructures have almost the same names  there are completely different types.
I didn't find a suitable go way to compare structs of different types that have fields of substructs of different types.

If there is one it will really help.
As I understood v1Profile and v1alphaV1 profile are completely different types that look the same.

Reply:
ok, makes sense as well